File: src/my_py_pkg/my_py_pkg/my_first_node.py
#! /usr/bin/env python3 

import rclpy
from rclpy.node import Node 

class MyCustomNode(Node):

    def __init__(self):
        super().__init__("my_custom_node")
        self.timer = self.create_timer(1.0, self.timer_callback)
        self.counter = 0 

    def timer_callback(self):
        self.get_logger().info("Hello ROS2 "+ str(self.counter))
        self.counter += 1
   
def main(args=None):
        rclpy.init(args=args)
        node = MyCustomNode()
        rclpy.spin(node)
        node.destroy_node()
        rclpy.shutdown()

if  __name__ == "__main__":
    main()

File: src/my_cpp_pkg/src/my_first_node.cpp
#include "rclcpp/rclcpp.hpp"


class MyFirstNode : public rclcpp::Node
{
public: 

MyFirstNode() : Node("my_first_node")
    {
        RCLCPP_INFO(this->get_logger(), "Hello, ROS2!");
    
    timer_ = this ->create_wall_timer(
        std::chrono::seconds(1),
        std::bind(&MyFirstNode::timer_callback, this)
    );
    }

private:
rclcpp::TimerBase::SharedPtr timer_;
void timer_callback()  
    {
        RCLCPP_INFO(this->get_logger(), "Hello, ROS2!");
    }


};

int main(int argc, char **argv)
{
    rclcpp::init(argc, argv);

    auto node = std::make_shared<MyFirstNode> ();


    rclcpp::spin(node);

    rclcpp::shutdown();

    return 0;
}